src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4723,3 +4735,249 @@ void C2_MacroAssembler::vector_reverse_byte(BasicType bt, XMMRegister dst, XMMRe
   }
   vpshufb(dst, src, dst, vec_enc);
 }
+
+void C2_MacroAssembler::vector_count_leading_zeros_evex(BasicType bt, XMMRegister dst, XMMRegister src,
+                                                        XMMRegister xtmp1, XMMRegister xtmp2, XMMRegister xtmp3,
+                                                        KRegister ktmp, Register rtmp, bool merge, int vec_enc) {
+  assert(is_integral_type(bt), "");
+  assert(VM_Version::supports_avx512vl() || vec_enc == Assembler::AVX_512bit, "");
+  assert(VM_Version::supports_avx512cd(), "");
+  switch(bt) {
+    case T_LONG:
+      evplzcntq(dst, ktmp, src, merge, vec_enc);
+      break;
+    case T_INT:
+      evplzcntd(dst, ktmp, src, merge, vec_enc);
```

Comment:
The ktmp here should be k0. An assert here or use explicit k0.

Reply:
ktmp will be a non k0 opmask register for predicated leading zero count only for INT/LONG type.   match_rule_supported_vector_masked carries necessary conditions.